src/ConfigOption.h

```diff
@@ -24,21 +24,33 @@ namespace Configuration {
 /// (i.e. the Config objects and similar/related global state). To facilitate
 /// reuse, implementations/specializations should also be independent from any
 /// specific configuration directive name and its squid.conf location.
-///
-/// TODO: Support multi-directive components of various kinds.
 template <class T>
 class Component
 {
 public:
-    /* the code adding "TYPE: T" to cf.data.pre must specialize these */
+    /* code adding "REPETITIONS: banned" to cf.data.pre must specialize these */
 
     /// creates a new T instance using the given parser; never returns nil
     static T Parse(ConfigParser &);
 
-    /// reports the current T instance configuration in squid.conf format
+    /// reports configuration parameters of the given T instance in squid.conf format
     static void Print(std::ostream &, const T &);
 
-    /// destroys Parse() result
+    /* code adding "REPETITIONS: update" to cf.data.pre must specialize these */
+
+    /// creates a new T instance for one or more ParseAndUpdate() calls
+    static T Create();
+
+    /// parses the first and any subsequent directive occurrence, updating T
+    static void ParseAndUpdate(T &, ConfigParser &);
+
+    // TODO: Upgrade to std::ostream after upgrading Security::PeerOptions::dumpCfg()
+    /// reports the given T instance configuration in squid.conf format
+    static void PrintDirectives(StoreEntry *, const T &, const char *directiveName);
```

Comment:
Q2: Should we upgrade PeerOptions::dumpCfg() and related code to use std::ostream (in a yet another separate PR)? 

Configuration::Components enhancement adds PrintDirectives() API that uses StoreEntry to report directives configuration. Its existing sister Print() API (for squid.conf directives that cannot be repeated) uses std::ostream. Ideally, the new API should use std::ostream as well, but we cannot do that easily because the existing PeerOptions::dumpCfg() code on that new API path still uses StoreEntry and has to be changed throughout to use std::ostream, introducing a lot of unrelated noise.

The alternative here is to add an API using StoreEntry and wait for a better opportunity to upgrade that (and all the new code that will start using that new API by then).

Reply:
From core meeting discussion:
 Changes included here are useful to simplify the feature addition, technical debt induced by incomplete transition is acceptable in light of how this overlaps the ongoing ConfigParser and CacheManager refactoring.

Reply:
> Should we upgrade PeerOptions::dumpCfg() and related code to use std::ostream (in a yet another separate PR)?

That upgrade is done at #1460.